net_transport.go

```diff
@@ -221,16 +221,34 @@ func (t *NetTransport) Shutdown() error {
 // and hands them off to the stream channel.
 func (t *NetTransport) tcpListen(tcpLn *net.TCPListener) {
 	defer t.wg.Done()
+
+	const baseDelay = 5 * time.Millisecond
+	const maxDelay = 1 * time.Second
+
+	var loopDelay time.Duration
 	for {
 		conn, err := tcpLn.AcceptTCP()
 		if err != nil {
 			if s := atomic.LoadInt32(&t.shutdown); s == 1 {
 				break
 			}
 
+			if loopDelay == 0 {
+				loopDelay = baseDelay
+			} else {
+				loopDelay *= 2
+			}
+
+			if loopDelay > maxDelay {
+				loopDelay = maxDelay
+			}
+
 			t.logger.Printf("[ERR] memberlist: Error accepting TCP connection: %v", err)
+			time.Sleep(loopDelay)
```

Comment:
The behaviour here is that we will delay `Shutdown` for up to `maxDelay` in case an accept error has just occured.

Given `maxDelay` is one second that seems fine and preferable to changing the shutdown mechanism to close a chan for example.

Is it worth adding a comment to the `maxDelay` const above just to note that assumption? e.g. in case anyone later thought it prudent to increase that value without noticing it would hard the shutdown responsiveness eve further?

Reply:
will add a comment. thanks!